deps/wazuh_testing/wazuh_testing/tools/monitoring.py

```diff
@@ -661,3 +669,196 @@ def queue(self):
 
         def put_queue(self, item):
             self._queue.put(item)
+
+
+def new_process(fn):
+    """Wrapper for enable multiprocessing inside a class
+
+    Parameters
+    ----------
+    fn : callable
+        Function to be executed in a new thread
+
+    Returns
+    -------
+    wrapper
+    """
+
+    def wrapper(*args, **kwargs):
+        thread = Process(target=fn, args=args, kwargs=kwargs)
+        thread.start()
+        return thread
+
+    return wrapper
+
+
+def callback_generator(regex):
+    def new_callback(line):
+        match = re.match(regex, line)
+        if match:
+            return line
+
+    return new_callback
+
+
+class HostMonitor:
```

Comment:
We could add a more general documentation to explain the main purpose of this class.